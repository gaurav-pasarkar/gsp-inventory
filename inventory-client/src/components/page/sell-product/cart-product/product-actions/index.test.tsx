import matchSnapshot from "../../../../../test-utils/match-snapshot";
import ProductActions from "./index";
import {fireEvent, render} from "@testing-library/react";

describe('Product actions', () => {

  const mockAdd = jest.fn();
  const mockIncrement = jest.fn();
  const mockDecrement = jest.fn();

  const props = {
    quantityInCart: 1,
    onAdd: mockAdd,
    onQuantityIncrement: mockIncrement,
    onQuantityDecrement: mockDecrement,
    isInCart: true
  }

  it('should render', () => {
    matchSnapshot(<ProductActions {...props}/>)
  });

  it('should render Add button if product is not in cart', () => {
    const {queryByText, queryByLabelText} = render(<ProductActions {...props} isInCart={false}/>);

    expect(queryByText(/Add/)).toBeInTheDocument();
    expect(queryByLabelText(/remove from shopping cart/)).not.toBeInTheDocument();
    expect(queryByLabelText(/add to shopping cart/)).not.toBeInTheDocument();
  });

  it('should render increment decrement button if product is in cart', () => {
    const {queryByText, queryByLabelText} = render(<ProductActions {...props} />);

    expect(queryByText(/Add/)).not.toBeInTheDocument();
    expect(queryByLabelText(/remove from shopping cart/)).toBeInTheDocument();
    expect(queryByLabelText(/add to shopping cart/)).toBeInTheDocument();
  });

  it('should call onQuantityIncrement when quantity is added', () => {
    const {getByLabelText} = render(<ProductActions {...props} />);

    fireEvent.click(getByLabelText(/add to shopping cart/));

    expect(mockIncrement).toHaveBeenCalled();
  });

  it('should call onQuantityDecrement when quantity is subtracted', () => {
    const {getByLabelText} = render(<ProductActions {...props} />);

    fireEvent.click(getByLabelText(/remove from shopping cart/));

    expect(mockDecrement).toHaveBeenCalled();
  });
});