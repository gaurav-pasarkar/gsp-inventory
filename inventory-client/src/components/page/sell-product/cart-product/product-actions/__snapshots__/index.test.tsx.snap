// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product actions should render 1`] = `
<div
  aria-label="outlined primary button group"
  className="MuiButtonGroup-root makeStyles-root-1"
  role="group"
>
  <button
    aria-label="remove from shopping cart"
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-textPrimary Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <p
        className="MuiTypography-root makeStyles-quantity-2 MuiTypography-body1 MuiTypography-alignCenter"
      >
        1
      </p>
    </span>
  </button>
  <button
    aria-label="add to shopping cart"
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </span>
  </button>
</div>
`;
